bundles/core/org.eclipse.smarthome.core.thing.test/src/test/java/org/eclipse/smarthome/core/thing/internal/ChannelItemProviderTest.java

```diff
@@ -120,6 +114,27 @@ public void testItemRemoval_itemFromOtherProvider() throws Exception {
         verify(listener, never()).added(same(provider), same(ITEM));
     }
 
+    @Test
+    public void testDisableBeforeDelayedInitialization() throws Exception {
+        provider = createProvider();
+
+        Map<String, Object> props = new HashMap<>();
+        props.put("enabled", "true");
+        provider.activate(props);
+
+        verify(linkRegistry, never()).addRegistryChangeListener(same(provider.linkRegistryListener));
+        verify(itemRegistry, never()).addRegistryHook(same(provider.itemRegistryListener));
+
+        props = new HashMap<>();
+        props.put("enabled", "false");
+        provider.modified(props);
+
+        Thread.sleep(2500);
```

Comment:
The test verifies that the methods are not called and that after 2.5 seconds they are still not called. I can't use waitForAssert, because waitForAssert is used if the assertion is not true at first but becomes true after some time.

Reply:
Yes, I feared that this is the case :( Though 2.5 seconds are rather long. Maybe while you are at it you can extract the magic "2" from `final long diff = System.nanoTime() - lastUpdate - TimeUnit.SECONDS.toNanos(2);` into a variable that you can configure through your test?

In any case this magic number should become at least a constant and not just appear deep inside the code. WDYT?

Reply:
I moved the number to a constant and in the test I'm using reflection to set it to a lower number so the test doesn't need to wait 2sec.